x-pack/plugins/security_solution/public/timelines/components/timeline/body/events/stateful_row_renderer/use_stateful_row_renderer.ts

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { EcsSecurityExtension } from '@kbn/securitysolution-ecs';
+import { useMemo } from 'react';
+import type { RowRenderer } from '../../../../../../../common/types';
+import { getRowRenderer } from '../../renderers/get_row_renderer';
+
+interface UseStatefulRowRendererArgs {
+  data: EcsSecurityExtension;
+  rowRenderers: RowRenderer[];
+}
+
+export function useStatefulRowRenderer(args: UseStatefulRowRendererArgs) {
+  const { data, rowRenderers } = args;
+  const rowRenderer = useMemo(() => getRowRenderer({ data, rowRenderers }), [data, rowRenderers]);
+
+  return {
```

Comment:
wouldn't we want to memoize this return object as well?

Reply:
You are right. Thanks for catching it. Done here :  886b161ef0e